shared/src/netlink.rs

```diff
@@ -23,14 +26,12 @@ fn if_nametoindex(interface: &InterfaceName) -> Result<u32, io::Error> {
 
 pub fn set_up(interface: &InterfaceName, mtu: u32) -> Result<(), io::Error> {
     let index = if_nametoindex(interface)?;
-    let message = LinkMessage {
-        header: LinkHeader {
-            index,
-            flags: IFF_UP,
-            ..Default::default()
-        },
-        nlas: vec![link::nlas::Nla::Mtu(mtu)],
-    };
```

Comment:
What was wrong with this original code?

Reply:
FWIW, I for one would have also preferred the previous style. It allows not having the variables `mut`, i.e. it encodes tiny bit more information in the source code.

Extracting nested structs to own variables is very fine, so the following would be also cool (but not advocating for it):
```rust
    let header = LinkHeader {
        index,
        flags: IFF_UP,
        ..Default::default()
    };
    let message = LinkMessage {
        header,
        nlas: vec![link::nlas::Nla::Mtu(mtu)],
    };
```

Reply:
I would also have preferred the previous style but they got rid of it, since they labeled all structs "non-exhuastive" to make their compatibility story simpler for them. Not my favorite tbh.

Reply:
Ahhh, non-exhaustive structs, I see.